core/templates/dev/head/domain/topic_viewer/TopicDataObjectFactory.ts

```diff
@@ -0,0 +1,86 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating instances of frontend topic-data domain
+ * objects.
+ */
+
+export class TopicData {
+  _topicName: string;
+  _topicId: string;
+  _canonicalStories: any;
+  _additionalStories: any;
+  _uncategorizedSkillIds: Array<string>;
+  _subtopics: any;
+  _degreesOfMastery: any;
```

Comment:
Ditto here and elsewhere.

In general, try not to use `any` keyword. If you have to use in worst case then try adding a comment explaining the reason for using `any`.